### PR TITLE
Add contrib layer for nim programming language

### PR DESCRIPTION
**Pull request by Gonzih**
Hello, this is small contrib layer for *.nim files that I use myself.

Most of stuff for this layer I took from go contrib layer. I'm newbies to spacemacs, so please if you see something that does not make sense point me to the resource that should help make things right.

Feedback is welcome!

Thanks!

### CODE REVIEW

File: contrib/!lang/nim/packages.el [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      '(company
+        company-nim
+        nim-mode))
+
+(defun nim/init-nim-mode ()
+  (use-package nim-mode
+    :defer t
+    :config
+    (progn
+      (defun spacemacs/nim-copmile-run ()
```
Comment (TheBB) [inline contrib/!lang/nim/packages.el]: Typo :-)

**Comment by Gonzih** [comment]
Thanks, @TheBB. Fixed, pushed with force to keep everything in one commit.

**Comment by TheBB** [comment]
Using `spacemacs|add-company-hook` and `company-backends-nim-mode` gives you some more control over which backends are enabled, which I find useful. I'm not sure why the go layer doesn't user them.

**Comment by aaroncm** [comment]
This should probably also remove nim-mode from the `extra-langs` layer, right?

**Comment by Gonzih** [comment]
@TheBB done!
@aaroncm done!

File: contrib/!lang/nim/packages.el (hunk 1 above)
Comment (Gonzih) [inline contrib/!lang/nim/packages.el]: fixed, thanks!

**Comment by TheBB** [comment]
Yeah, looks good. You will still need a `nim/init-company-nim` function though, otherwise the package won't be installed or loaded.

:+1: otherwise

**Comment by Gonzih** [comment]
@TheBB thanks, is this better now?

**Comment by TheBB** [comment]
Yep, perfect.

**Comment by Gonzih** [comment]
Yay, thanks guys!

**Comment by trishume** [comment]
:+1:

**Comment by robbyoconnor** [comment]
Shit no flycheck support -- Write it if you have the time :D

**Comment by Gonzih** [comment]
@robbyoconnor I added flycheck-nim, but not sure why it does not work.

**Comment by robbyoconnor** [comment]
@Gonzih -- That was a fun adventure -- you have to figure out the regex pattern of the errors and write it in rx.el format -- look at the flycheck docs -- they do cover this -- I mean without it's not a problem -- I personally have no stake in it...I wrote the jade one because I wanted it but yeh I can help you if you want  -- use http://regex101.com -- it's an awesome site..if you're fine without -- then just leave it as-is :)

**Comment by TheBB** [comment]
@robbyoconnor If you read the code you would see he's trying to install a syntax checker someone else has already written. Clearly no need to mess around with flycheck and regular expressions here.

@Gonzih You should add `flycheck-nim` as an extension and not a package. Put `flycheck-nim` in `yourlayer/extensions/flycheck-nim/flycheck-nim.el`, and intialize it in `extensions.el` instead of `packages.el`.

**Comment by Gonzih** [comment]
@thebb I added flycheck-nim to extensions, not sure about initialization though. How do I load flycheck-nim only after flycheck-mode is initialized? How can I do that in packages.el? Thank you for your time and patience :)

**Comment by TheBB** [comment]
I made a couple of comments which I _think_ ought to do it.

**Comment by robbyoconnor** [comment]
Ah @TheBB -- then @Gonzih -- forget what I said <3

**Comment by robbyoconnor** [comment]
Why not contribute this back to flycheck -- or better yet why didn't the author do that :/

**Comment by Gonzih** [comment]
@robbyoconnor i asked author to publish library to the repository, once it's there i will change my implementation. also it might at some point become part of nim-mode (which is already contains lot of things).

@TheBB thanks, i see your comments.

**Comment by Gonzih** [comment]
Yay, it finally works, thank you guys! You are awesome :)

**Comment by Gonzih** [comment]
What do I need to do to get this patch merged? Just patiently wait? :)

**Comment by robbyoconnor** [comment]
@Gonzih -- I bet if you pester @syl20bnr enough it will happen...either that or he'll grow to hate you...

**Comment by mbertheau** [comment]
The correct answer is wait patiently :)

**Comment by robbyoconnor** [comment]
Many of us (including myself) have PRs outstanding -- it's one man doing all of this...give him a break...if you want a layer -- you can always merge it local -- I [wrote a blog post](robby.oconnor.ninja/2015/03/how-to-checkout-github-pull-requests.html) about how to check out pull requests locally using CLI git :)

**Comment by syl20bnr** [comment]
Thank you :+1: 
Cherry-picked into `develop` branch.

**Comment by Gonzih** [comment]
Cool! Thanks a lot!

**Pull request closed without merging**